### ISSUE TITLE
TLD > BCP 47 function?

### ISSUE THREAD

**Issue by binarykitchen** [wooorm/bcp-47#17]
Probably not scope of your package but I am longing for a function to return me a BCP 47 language tag based by a top level domain (TLD).

Have initially reported the need in https://github.com/petert82/country-query/issues/3 but since it seems inactive, and yours comes - after some research - the most close, I wonder why not ask you about that?

**Comment by wooorm** [comment]
Interesting problem! And hi, Michael 👋

First: this should definitely be another project. Maybe you could do a map of TLDs to BCP-47 tags, in JSON, but it should be outside `bcp-47`.

Second: There are quite some TLDs (generic, test, infrastructure) that do not map to anything related to your problem.

Third: TLDs are for countries (they’re often based on ISO 3166-1). BCP-47 is for languages. Some languages come with some geography information (e.g., `en-GB` for English as spoken in Greater Britain). But, regions could be added to any language. `nl-GB`, which would stand for Dutch as spoken  in GB, is valid.

To recap the biggest problem with your question: the `.nl` TLD (Netherlands), does not say anything about the site being in Dutch. Sites could be (and often are) in English or Frisian. Or anything else really.
Although Dutch is indeed the most common language in The Netherlands, not all regions have a favourite language.

**Comment by binarykitchen** [comment]
hey dude

thanks for the good feedback. totally agree in all points. let me explain the use case i have here to crack that problem:

on www.videomail.io (try it out), users can enter recipient email addresses. when recipient views the videomail, the sent date is shown but always in the wrong format. hence i am thinking of a tld to bcp 47 converter for better date display. tld is based on the recipient's email address. that new project could be called `email-to-bcp47` or `tld-to-bcp47` whatever. 

totally aware that this isn't entirely accurate (see your dutch example) but it is still better than always displaying the hardcoded date format 

thoughts?

**Comment by wooorm** [comment]
I’d suggest using the `Date` JavaScript object to infer timezones (see [`moment`](http://momentjs.com), [`moment-timezone`](http://momentjs.com/timezone/)), and detecting the language the user prefers (see [`locale2`](https://github.com/moimikey/locale2)) instead!

I’d sa it’s a bad idea to do anything with bcp-47, and anyway, it should be a separate package!
